import "ydb/public/api/protos/ydb_issue_message.proto";
import "ydb/public/api/protos/ydb_operation.proto";
import "ydb/public/api/protos/ydb_status_codes.proto";

import "google/protobuf/timestamp.proto";

package NKikimrForcedCompaction;
option java_package = "ru.yandex.kikimr.proto";

message TForcedCompaction {
    optional uint64 Id = 1;
    optional Ydb.StatusIds.StatusCode Status = 2;
    repeated Ydb.Issue.IssueMessage Issues = 3;
    optional uint64 TotalShards = 4;
    optional uint64 CompactedShards = 5;
    optional uint64 QueuedShards = 6;
    optional google.protobuf.Timestamp StartTime = 7;
    optional google.protobuf.Timestamp EndTime = 8;
    optional string UserSID = 9;
}

message TCreateForcedCompactionRequest {
    optional Ydb.Operations.OperationParams OperationParams = 1;
    optional uint64 OwnerPathId = 2;
    optional uint64 LocalPathId = 3;
}

message TEvCreateForcedCompactionRequest {
    optional uint64 TxId = 1;
    optional string DatabaseName = 2;
    optional TCreateForcedCompactionRequest Request = 3;
    optional string UserSID = 4;
    optional string PeerName = 5;
}

message TCreateForcedCompactionResponse {
    optional TForcedCompaction Entry = 1;
}

message TEvCreateForcedCompactionResponse {
    optional uint64 TxId = 1;
    optional TCreateForcedCompactionResponse Response = 2;
}

message TGetForcedCompactionRequest {
    optional uint64 Id = 1;
}

message TEvGetForcedCompactionRequest {
    optional string DatabaseName = 2;
    optional TGetForcedCompactionRequest Request = 1;
}

message TGetForcedCompactionResponse {
    optional TForcedCompaction Entry = 1;
}

message TEvGetForcedCompactionResponse {
    optional TGetForcedCompactionResponse Response = 1;
}
